connect/runtime/src/main/java/org/apache/kafka/connect/runtime/WorkerSinkTask.java

```diff
@@ -695,9 +705,28 @@ private class HandleRebalance implements ConsumerRebalanceListener {
         @Override
         public void onPartitionsAssigned(Collection<TopicPartition> partitions) {
             log.debug("{} Partitions assigned {}", WorkerSinkTask.this, partitions);
-
+            Set<String> deletedTopics = new HashSet<>();
             for (TopicPartition tp : partitions) {
-                long pos = consumer.position(tp);
+                if (deletedTopics.contains(tp.topic())) {
+                    log.debug("Not assigning offsets for topic-partition {} since the topic {} has been deleted", tp, tp.topic());
+                    continue;
+                }
+                long pos;
+                try {
+                    pos = consumer.position(tp);
+                } catch (TimeoutException e) {
+                    log.error("TimeoutException occurred when fetching position for topic partition {}. " +
+                            "Checking if the topic {} exists", tp, tp.topic());
+                    Map<String, TopicDescription> topic = topicAdmin.describeTopics(tp.topic());
```

Comment:
> This adds new ACL requirements for sink connectors' admin clients,

That's a good point and something I missed. 

> Probably safest to assume that the topic exists if we fail to describe it.

Yeah.. In the case when let's say `TopicAuthorizationException` is thrown, it again becomes difficult to say if the topic exists or not. The `TopicAdmin#describeTopics` throws a `ConnectException` in this case which could be a false positive. And since this PR throws any exceptions from `TopicAdmin#describeTopics` the connector might now fail with ConnectException, albeit a different one. So, as you said might be safe to assume that topic exists then.